src/ray/gcs/gcs_client/accessor.cc

```diff
@@ -173,6 +173,24 @@ Status ActorInfoAccessor::AsyncGetAllByFilter(
     const std::optional<std::string> &actor_state_name,
     const MultiItemCallback<rpc::ActorTableData> &callback,
     int64_t timeout_ms) {
+  return AsyncRawGetAllActorInfo(
+      actor_id,
+      job_id,
+      actor_state_name,
+      /*limit=*/-1,
+      [callback](const Status &status, std::optional<rpc::GetAllActorInfoReply> &&reply) {
+        callback(status, VectorFromProtobuf(reply->actor_table_data()));
```

Comment:
Do we need to consider `reply` being nullptr case since it's `std::optional`?

Reply:
no need, since the GcsRpcClient always gives a reply (empty on error). See line 213.

Reply:
why is it optional in this case?